packages/roosterjs-editor-plugins/lib/plugins/AutoFormat/AutoFormat.ts

```diff
@@ -0,0 +1,86 @@
+import {
+    EditorPlugin,
+    ExperimentalFeatures,
+    IEditor,
+    PluginEvent,
+    PluginEventType,
+    PositionType,
+} from 'roosterjs-editor-types';
+
+const specialCharacters = /[`!@#$%^&*()_+\=\[\]{};':"\\|,.<>\/?~]/;
+
+/**
+ * Automatically transform -- into hyphen, if typed between two words.
+ */
+export default class AutoFormat implements EditorPlugin {
+    private editor: IEditor;
+    private lastKeyTyped: string;
+
+    /**
+     * Get a friendly name of this plugin
+     */
+    getName() {
+        return 'AutoFormat';
+    }
+
+    /**
+     * Initialize this plugin
+     * @param editor The editor instance
+     */
+    initialize(editor: IEditor) {
+        this.editor = editor;
+    }
+
+    /**
+     * Dispose this plugin
+     */
+    dispose() {
+        this.editor = null;
+    }
+
+    /**
+     * Handle events triggered from editor
+     * @param event PluginEvent object
+     */
+    onPluginEvent(event: PluginEvent) {
+        if (
+            event.eventType === PluginEventType.KeyDown &&
+            this.editor.isFeatureEnabled(ExperimentalFeatures.AutoHyphen)
```

Comment:
No need to check feature. If we don't want this feature, just don't add this plugin at all.